crates/kornia-io/src/jpeg.rs

```diff
@@ -106,10 +127,31 @@ fn read_image_jpeg_internal<const N: usize>(
     Ok(Image::new(image_size, img_data)?)
 }
 
+fn decode_jpeg_impl<const C: usize>(image: &mut Image<u8, C>, bytes: &[u8]) -> Result<(), IoError> {
+    let mut decoder = zune_jpeg::JpegDecoder::new(bytes);
+    decoder
+        .decode_headers()
+        .map_err(IoError::JpegDecodingError)?;
+
+    let image_info = decoder.info().ok_or_else(|| {
+        IoError::JpegDecodingError(zune_jpeg::errors::DecodeErrors::Format(String::from(
+            "Failed to find image info from its metadata",
+        )))
+    })?;
+
+    let image_size = [image_info.height as usize, image_info.width as usize, C];
+    decoder.decode_into(image.as_slice_mut())?;
+
+    // Update the tensor shape and stride
+    image.0.shape = image_size;
```

Comment:
i don't think it's necessary if the image has been properly allocated. You should check sizes and return error before